pkg/cmd/set/set_image_test.go

```diff
@@ -612,6 +614,22 @@ func TestSetImageRemote(t *testing.T) {
 			path:         "/namespaces/test/replicationcontrollers/nginx",
 			args:         []string{"replicationcontroller", "nginx", "*=thingy"},
 		},
+		{
+			name: "set image kruiseappsv1alpha1.SidecarSet",
+			object: &kruiseappsv1alpha1.SidecarSet{
+				ObjectMeta: metav1.ObjectMeta{Name: "nginx"},
+				Spec: kruiseappsv1alpha1.SidecarSetSpec{
+					Containers: []kruiseappsv1alpha1.SidecarContainer{
+						{
+							Container: corev1.Container{Name: "nginx", Image: "nginx"},
+						},
+					},
+				},
+			},
+			groupVersion: corev1.SchemeGroupVersion,
+			path:         "/namespaces/test/sidecarsets/nginx",
```

Comment:
since sidecarset is a cluster scoped resource, i doubt whether the path test is valid here

Reply:
Because the SidecarSet is a cluster-wide resource, it is defined across all namespaces and can be operated on by all namespaces, even the namespaces does not exist.

This can work
```shell
# dummy-ns do not existed
[root@zone-a code]# kubectl get sidecarset -n dummy-ns
NAME              MATCHED   UPDATED   READY   AGE
test-sidecarset   1         1         1       28m
```

This also can work
```shell
# dummy-ns do not existed
kubectl kruise set image sidecarsets/test-sidecarset sidecar1=debian:11 -n chengleqi -n dummy-ns
# output sidecarset.apps.kruise.io/test-sidecarset image updated
```